### PR TITLE
PLFM-4317 - refactor

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java [hunk 1]
```diff
@@ -212,18 +212,35 @@ public SubscriptionPagedResults getAll(String subscriberId, Long limit,
...
-			ValidateArgument.required(projectIds, "projectIds");
-			if (projectIds.isEmpty()) {
-				results.setResults(new ArrayList<Subscription>());
-				results.setTotalNumberOfResults(0L);
-				return results;
-			}
-			parameters.addValue("projectIds", projectIds);
+		String query = SQL_GET_ALL_SUB;
+		String countQuery = getCountQuery(query);
+		results.setTotalNumberOfResults(namedTemplate.queryForObject(countQuery, parameters, Long.class));
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java:217]: can we change this to a next page token?

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java [hunk 2]
```diff
@@ -233,26 +250,40 @@ public SubscriptionPagedResults getAll(String subscriberId, Long limit,
...
+			results.setResults(new ArrayList<Subscription>());
+			results.setTotalNumberOfResults(0L);
+			return results;
 		}
+
+		MapSqlParameterSource parameters = new MapSqlParameterSource("subscriberId", subscriberId);
+		parameters.addValue("projectIds", projectIds);
+
+		String query = SQL_GET_ALL_FORUM_SUB;
+		String countQuery = getCountQuery(query);
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java:272]: again, next page token.

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java [hunk 3]
```diff
@@ -262,29 +293,50 @@ public SubscriptionPagedResults getSubscriptionList(String subscriberId,
...
-				return SQL_GET_FORUM_SUB_LIST;
-			case THREAD:
-				return SQL_GET_THREAD_SUB_LIST;
-			case DATA_ACCESS_SUBMISSION_STATUS:
-				return SQL_GET_SUB_LIST;
-			default:
-				throw new RuntimeException("Unsupported type "+objectType.name());
+	@Override
+	public SubscriptionPagedResults listSubscriptionForThread(String subscriberId, List<String> ids) {
+		ValidateArgument.required(subscriberId, "subscriberId");
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java:308]: Why does this return a paginated result when it takes a list of ID and does not include limit or offset?

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java [hunk 4]
```diff
@@ -262,29 +293,50 @@ public SubscriptionPagedResults getSubscriptionList(String subscriberId,
...
+			parameters.addValue("subscriberId", subscriberId);
+			List<Subscription> subscriptions = namedTemplate.query(SQL_GET_THREAD_SUB_LIST, parameters, ROW_MAPPER);
+			results.setResults(subscriptions);
+			results.setTotalNumberOfResults((long) subscriptions.size());
+		}
+		return results;
+	}
+
+	@Override
+	public SubscriptionPagedResults listSubscriptionForForum(String subscriberId, List<String> ids) {
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java:325]: Same not a paginated result.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/subscription/SubscriptionManagerImpl.java [hunk 5]
```diff
@@ -67,7 +67,16 @@ public SubscriptionPagedResults getList(UserInfo userInfo, SubscriptionRequest r
 		ValidateArgument.required(request, "request");
 		ValidateArgument.required(request.getObjectType(), "SubscriptionRequest.objectType");
 		ValidateArgument.required(request.getIdList(), "SubscriptionRequest.idList");
-		return subscriptionDao.getSubscriptionList(userInfo.getId().toString(), request.getObjectType(), request.getIdList());
+		switch(request.getObjectType()) {
+		case THREAD:
+			return subscriptionDao.listSubscriptionForThread(userInfo.getId().toString(), request.getIdList());
+		case FORUM:
+			return subscriptionDao.listSubscriptionForForum(userInfo.getId().toString(), request.getIdList());
+		case DATA_ACCESS_SUBMISSION_STATUS:
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/subscription/SubscriptionManagerImpl.java:75]: I thought thread and forum where the special cases and all others could use the generic list subscriptions.  Should the default call the generic list subscriptions?

**Comment by john-hill** [comment]
Thank you for moving the logic to the manager!

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/subscription/SubscriptionManagerImpl.java (hunk 5 above)
Comment (kimyen) [reply services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/subscription/SubscriptionManagerImpl.java:75]: listSubscriptions does not work for DATA_ACCESS_SUBMISSION or any SubscriptionObjectType that users can subscribeAll to (in the future). This switch captured the types that we support.

**Pull request merged**